### ISSUE TITLE
Algorithm leaderboard and cards design

### ISSUE THREAD

**Issue by amickan** [comic/grand-challenge.org#2109]
**Algorithm cards**

It would be nice to add a few links to the algorithm cards (akin to how we do this for challenge cards, with badges):
- link to linked github repo if there is one
- link to linked publication(s)
- link to the challenge(s) this algorithm was submitted to 

The latter is quite a complex lookup: I think it requires accessing the submission set associated with the algorithm's most recent container image and then retrieving the challenge each of these submissions is associated with (through the phase they are associated with). We would need to add a prefetch_related call to the algorithm list view to save on look-ups.

**Comment by jmsmkn** [comment]
Yes, the link to challenges is going to be too slow. It would be a join across 4 tables (`algorithm_image_set__submission_set__phase__challenge`) plus a distinct on a page that is already over budget.
